Linux-0.98/Yggdrasil-0.98.3/usr/src/TeX/transfig/doc/transfig.1
.TH TRANSFIG 1 "1 Sept 1990"
.SH NAME
transfig \- creates a makefile for portable LaTeX figures
.SH SYNOPSIS
.B transfig
[ -L \fIlanguage\fR ]
[ -M \fImakefile\fR ]
[ -T \fItexfile\fR ]
[ -I \fImacrofile\fR ]
[ -V ] 
[ -a ] 
[ \fIcontrol_specs\fR figfile ... ] ...]
.SH DESCRIPTION
.I Transfig
creates a makefile to translate figures described in Fig code, PostScript,
or PIC into a specified LaTeX graphics \fIlanguage\fR.
PostScript and PIC files are identified by the suffixes ".ps" and ".pic"
respectively;
Fig files can be specified either with or without the suffix ".fig" .
Transfig also creates a TeX macro file appropriate to the target language.
.SH OPTIONS
The \fIlanguage\fR specifier must be either
.T
.BR epic ,
.BR eepic ,
.BR eepicemu ,
.BR latex ,
.BR pictex ,
.BR ps ,
.BR psfig ,
.BR pstex ,
.BR textyl ,
.BR tpic ,
or
.BR null .
The specifier
.B psfig
is like the 
.B ps
specifier, except that it
make use of the psfig macro package.
The specifier
.B pstex
is like
.B psfig,
except that it uses LaTeX to process formatted (special) text.
The specifier 
.B null
results in the figures being replaced by empty boxes of the
appropriate size.
If no language is specified,
.B epic
is the default.
.PP
The names of the makefile and TeX macro file can be set with the
-M and -T options.
The default names are "Makefile" and "transfig.tex", respectively.
If the -I option is specified, then a command to read in \fImacrofile\fR is
	inserted into the TeX macro file.
If the -V option is specified, the program version number is printed only.
.PP
Control specifiers are 
.TP
\fB\-m\fI mag\fR
set the magnification at which the figure is rendered (default 1.0).
.TP
\fB\-f\fI font\fR
set the default font family used for text objects (default "\fBcmr\fR").
.TP
\fB\-s \fIfontsize\fR
set the default font size (in points) for text objects (default 11*\fImag\fR).
.TP
\fB\-a\fR
use alternate font encoding scheme.
.TP
\fB\-o \fIoption\fR
pass the specified option string through to fig2dev.
.PP
If there is no scaling, the default font is "\fB\\elvrm\fR".
Language and control specifiers may occur more than once;
they affect the translation of files which appear to their
right in the argument list. 
If no files are specified, then "*.fig" is assumed.
.SH "SEE ALSO"
fig(1),
fig2dev(1),
pic(1),
pic2fig(1),
.SH AUTHOR
Micah Beck


Linux-0.98/Yggdrasil-0.98.3/usr/src/TeX/transfig/transfig/transfig.c
/*
 * TransFig: Facility for Translating Fig code
 * Copyright (c) 1991 Micah Beck, Cornell University
 *
 * Permission to use, copy, modify, distribute, and sell this software and its
 * documentation for any purpose is hereby granted without fee, provided that
 * the above copyright notice appear in all copies and that both that
 * copyright notice and this permission notice appear in supporting
 * documentation, and that the name of Cornell University not be used in
 * advertising or publicity pertaining to distribution of the software without
 * specific, written prior permission.  Cornell University makes no
 * representations about the suitability of this software for any purpose.  It
 * is provided "as is" without express or implied warranty.
 *
 * CORNELL UNIVERSITY DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE,
 * INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO
 * EVENT SHALL CORNELL UNIVERSITY BE LIABLE FOR ANY SPECIAL, INDIRECT OR
 * CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE,
 * DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER
 * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 * PERFORMANCE OF THIS SOFTWARE.
 *
 */

/*
 * transfig: 	figure translation setup program
 *		creates TeX macro file and makefile
 *
 * usage: transfig <option> ... [[<flag> ... ] [<figure>] ... ] ...
 *
 * where:	<option> = -L <language> | -M <makefile> | -T <texfile>
 *		<flag>	 = -f <font> | -s <size> | -m <scale>
 */

#include <stdio.h>
#include "patchlevel.h"
#include "transfig.h"


argument *parse_arg(), *arglist = NULL, *lastarg = NULL;
char *strip();

char *mkfile = "Makefile";
char *txfile = "transfig.tex";
char *input = "";
int  altfonts = 0;

char *lname[] = {
	"box",
	"epic",
	"eepic",
	"eepicemu",
	"latex",
	"pictex",
	"ps",
	"psfig",
	"pstex",
	"textyl",
	"tpic"};

/* enum input {apg, fig, pic, ps}; */
char *iname[] = {
	"apg",
	"fig",
  	"pic",
	"ps"};
 
main(argc, argv)
int argc;
char *argv[];
{
  FILE *mk, *tx;
  enum language tolang = epic;
  argument *a;
  char c, *cp; 
  char *arg_f = NULL, *arg_s = NULL, *arg_m = NULL, *arg_o = NULL, *argbuf;

  for ( optind = 1; optind < argc; optind++ ) {
    cp = argv[optind];
    if (*cp == '-')
    {
  	if (!cp[1])
	{
		fprintf(stderr, "transfig: bad option format '-'\n");
		exit(1);
	}
	if (cp[1] == 'V') 
	{
		fprintf(stderr, "TransFig Version %s Patchlevel %s\n",
							VERSION, PATCHLEVEL);
		exit(0);
	}
	if (cp[2])
	{
		optarg = &cp[2];
	} else
	{
		if (cp[1] != 'a') {
			optind += 1;
			if (optind == argc)
			{
				fprintf(stderr,
					"transfig: no value for '%c' arg\n", cp[1]);
				exit(1);
			}
			optarg = argv[optind];
		}
	}
 	switch (cp[1]) {

	case 'I':
		input = optarg;
		break;

  	case 'L':
		tolang = str2lang(optarg);
		break;
  	case 'M':
		mkfile = optarg;
		break;
  	case 'T':
		txfile = optarg;
		break;
	case 'a':
		altfonts = 1;
		break;
	case 'f':
		arg_f = optarg;
		break;
	case 's':
		arg_s = optarg;	
		break;
	case 'm':
		arg_m = optarg;	
		break;

	case 'o':
		arg_o = optarg;
		break;

  	default:
		fprintf(stderr, "transfig: illegal option -- '%c'\n", cp[1]);
		exit(1);
  	}
    } else
    {
	a = parse_arg(tolang, arg_f, arg_s, arg_m, arg_o, argv[optind]);

	if ( !lastarg )
		arglist = a;
	else
		lastarg->next = a; 
	lastarg = a;
    }
  }

  /* no files specified -> all files */
  if (!arglist)
  {
	argbuf = sysls();
	while (cp = strchr(argbuf, '\n'))
	{
		*cp = '\0';
		a = parse_arg(tolang, arg_f, arg_s, arg_m, arg_o, argbuf);
		if ( !lastarg )
			arglist = a;
		else
			lastarg->next = a; 
		lastarg = a;
		argbuf = cp+1;
	}
  }

  sysmv(txfile);
  tx = fopen(txfile, "w");
  texfile(tx, input, altfonts, arglist);

  sysmv(mkfile);
  mk = fopen(mkfile, "w");
  makefile(mk, altfonts, arglist);
  exit(0);
}

enum language str2lang(s)
char *s;
{
  int i;

  /* aliases */
  if (!strcmp(s, "pic")) return tpic;
  if (!strcmp(s, "postscript")) return postscript;
  if (!strcmp(s, "latexps")) return pstex;
  if (!strcmp(s, "null")) return box;

  /* real names*/
  for (i = 0; i <= (int)MAXLANG; i++)
	if (!strcmp(lname[i], s)) return (enum language)i;

  /* other strings */
  fprintf(stderr, "Unknown output language \"%s\"\n", s);
  exit(1);
}

argument *parse_arg(tolang, arg_f, arg_s, arg_m, arg_o, arg)
enum language tolang;
char *arg_f, *arg_s, *arg_m, *arg_o, *arg;
{
  argument *a;

  a = (argument *)malloc(sizeof(argument));
  a->f = arg_f;
  a->s = arg_s;
  a->m = arg_m;
  a->o = arg_o;
  a->next = NULL;
  a->tofig = NULL;
  a->topic = NULL;
  a->tops = NULL;
  a->tolang = tolang;
  
  /* PIC */
  if (strip(arg, ".pic"))
  {
  	a->name = mksuff(arg, "");
  	a->type = pic;
	a->tofig = PIC2FIG;
	return a;
  }

  /* PS format */
  if (strip(arg, ".ps"))
  {
  	a->name = mksuff(arg, "");
  	a->type = ps;
 	return a;
  }

  /* ApGraph format */
  if (strip(arg, ".apg"))
  {
  	a->name = mksuff(arg, "");
  	a->type = apg;
	a->tofig = APG2FIG;
 	return a;
  }

  /* Fig format */
  strip(arg, ".fig");
  a->name = mksuff(arg, "");
  a->type = fig;
  return a;
}
